Makefile

```diff
@@ -119,3 +119,20 @@ update-dep:
 .PHONY: update-otel
 update-otel:
 	$(MAKE) update-dep MODULE=go.opentelemetry.io/collector VERSION=master
+
+.PHONY: otel-from-tree
+otel-from-tree:
+	# This command allows you to make changes to your local checkout of otel core and build
+	# contrib against those changes without having to push to github and update a bunch of
+	# references. The workflow is:
+	#
+	# 1. Hack on changes in core (assumed to be checked out in ../opentelemetry-collector from this directory)
+	# 2. Run `make otel-from-tree` (only need to run it once to remap go modules)
+	# 3. You can now build contrib and it will use your local otel core changes.
+	# 4. Before committing/pushing your contrib changes, undo by running `make otel-from-lib`.
+	$(MAKE) for-all CMD="go mod edit -replace go.opentelemetry.io/collector=$(SRC_ROOT)/../opentelemetry-collector"
```

Comment:
Can we parameterize `../opentelemetry-collector` part but keep it as the default value?